From 7b0893d8198a2b8454d9757c1b92da40478c2d31 Mon Sep 17 00:00:00 2001
From: Ewan Milne <emilne@redhat.com>
Date: Wed, 20 Aug 2014 18:59:09 -0400
Subject: [scsi] Implement ch_printk()

Message-id: <1408561176-18133-31-git-send-email-emilne@redhat.com>
Patchwork-id: 88446
O-Subject: [RHEL7.1 PATCH e-stor 30/57] scsi: Implement ch_printk()
Bugzilla: 998695
RH-Acked-by: David Milburn <dmilburn@redhat.com>
RH-Acked-by: Tony Camuso <tcamuso@redhat.com>

From: "Ewan D. Milne" <emilne@redhat.com>

upstream commit b89160719468b9dc57222a632a4ca7e49906554b

Tested by Ewan D. Milne and Barry Donahue with a combination
of manual and automated testing.  No issues seen.

upstream commit b89160719468b9dc57222a632a4ca7e49906554b
Author: Hannes Reinecke <hare@suse.de>
Date:   Wed Jun 25 16:39:56 2014 +0200

    scsi: Implement ch_printk()

    Update the ch driver to use dev_printk() variants instead of
    plain printk(); this will prefix logging messages with the
    appropriate device.

    Signed-off-by: Hannes Reinecke <hare@suse.de>
    Reviewed-by: Martin K. Petersen <martin.petersen@oracle.com>
    Reviewed-by: Christoph Hellwig <hch@lst.de>
    Signed-off-by: Christoph Hellwig <hch@lst.de>

Signed-off-by: Ewan D. Milne <emilne@redhat.com>
Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/drivers/scsi/ch.c b/drivers/scsi/ch.c
index 2a32374..ba1c3bf 100644
--- a/drivers/scsi/ch.c
+++ b/drivers/scsi/ch.c
@@ -84,15 +84,19 @@ static const char * vendor_labels[CH_TYPES-4] = {
 };
 // module_param_string_array(vendor_labels, NULL, 0444);
 
+#define ch_printk(prefix, ch, fmt, a...) \
+ sdev_printk(prefix, (ch)->device, "[%s] " fmt, \
+      (ch)->name, ##a)
+
 #define DPRINTK(fmt, arg...)      \
 do {         \
  if (debug)       \
-  printk(KERN_DEBUG "%s: " fmt, ch->name, ##arg);  \
+  ch_printk(KERN_DEBUG, ch, fmt, ##arg);   \
 } while (0)
 #define VPRINTK(level, fmt, arg...)     \
 do {         \
  if (verbose)       \
-  printk(level "%s: " fmt, ch->name, ##arg);  \
+  ch_printk(level, ch, fmt, ##arg);   \
 } while (0)
 
 /* ------------------------------------------------------------------- */
@@ -196,7 +200,7 @@ ch_do_scsi(scsi_changer *ch, unsigned char *cmd,
   __scsi_print_command(cmd);
  }
 
-        result = scsi_execute_req(ch->device, cmd, direction, buffer,
+ result = scsi_execute_req(ch->device, cmd, direction, buffer,
       buflength, &sshdr, timeout * HZ,
       MAX_RETRIES, NULL);
 
@@ -924,8 +928,8 @@ static int ch_probe(struct device *dev)
       MKDEV(SCSI_CHANGER_MAJOR, ch->minor), ch,
       "s%s", ch->name);
  if (IS_ERR(class_dev)) {
-  printk(KERN_WARNING "ch%d: device_create failed\n",
-         ch->minor);
+  sdev_printk(KERN_WARNING, sd, "ch%d: device_create failed\n",
+       ch->minor);
   ret = PTR_ERR(class_dev);
   goto remove_idr;
  }
-- 
1.7.1